src/ChunkMap.cpp

```diff
@@ -2718,25 +2719,69 @@ void cChunkMap::TickBlock(int a_BlockX, int a_BlockY, int a_BlockZ)
 
 
 
-void cChunkMap::UnloadUnusedChunks(void)
+int cChunkMap::UnloadUnusedChunks(int a_Amount)
 {
+	int UnloadedTotal = 0;
+	bool UnloadAll;
+
+	if (a_Amount == 0)
+	{
+		UnloadAll = false;
+		a_Amount = m_World->GetNumChunks();
+	}
+	else
+	{
+		UnloadAll = true;
+	}
+
 	cCSLock Lock(m_CSLayers);
-	for (cChunkLayerList::iterator itr = m_Layers.begin(); itr != m_Layers.end(); ++itr)
+	if (!UnloadAll)
 	{
-		(*itr)->UnloadUnusedChunks();
-	}  // for itr - m_Layers
+		for (auto Layer : m_Layers)
+		{
+			Layer->UnloadUnusedChunks(a_Amount);
+		}
+		return a_Amount;
+	}
+	else
+	{
+		cFastRandom Rand;
+		int Size = static_cast<int>(m_Layers.size());
+		// Choose 10 layers randomly and unload them.
```

Comment:
This is not the most optimal technique, but given the current data structures, it is the simplest. It might be worth modifying those data structures for squeezing better performance, though.